src/kafka/consumer.clj

```diff
@@ -0,0 +1,61 @@
+(ns kafka.consumer
+  (:gen-class)
+  (:import
+    (java.time Duration)
+    (java.net InetAddress)
+    (org.apache.kafka.common.serialization StringDeserializer)
+    (org.apache.kafka.common.errors WakeupException)))
+
+(def consumer-properties
+  {"client.id",         (.getHostName (InetAddress/getLocalHost))
+   "group.id",          "group1"
+   "bootstrap.servers", "localhost:9092"
+   "key.deserializer"   StringDeserializer
+   "value.deserializer" StringDeserializer})
+
+(defn consumer-subscribe
+  [consumer topic]
+  (.subscribe consumer [topic])
+  (.commitAsync consumer))
+
+(defn receive-msg-single-thread
+  [consumer-properties]
+  (let [consumer (KafkaConsumer. consumer-properties)
+        topic    "example-topic"]
+    (consumer-subscribe consumer topic)
+    (while true
+      (let [records (.poll consumer (Duration/ofMillis 1))]
+        (doseq [record records]
+          (println "Message: " (.value record)))))))
```

Comment:
Only need 1 fn to demonstrate receiving messages. That fn should follow basic data-oriented principles. 
* Why does this fn block?
* Why does this fn not return data?

Reply:
_Poor_ decomposition. Why is everything being slammed together in one function? For example, how is building a consumer related to polling for records?